tests/test_utils/test_compat_conig.py

```diff
@@ -0,0 +1,120 @@
+    cfg = ConfigDict(dict(data=dict(val=dict(), test=dict(), train=dict())))
+    cfg = compat_loader_args(cfg)
+    # auto fill loader args
+    assert 'val_dataloader' in cfg.data
+    assert 'train_dataloader' in cfg.data
+    assert 'test_dataloader' in cfg.data
+    cfg = ConfigDict(
+        dict(
+            data=dict(
+                samples_per_gpu=1,
+                persistent_workers=True,
+                workers_per_gpu=1,
+                val=dict(samples_per_gpu=3),
+                test=dict(samples_per_gpu=2),
+                train=dict())))
+    with pytest.warns(None) as record:
+        cfg = compat_loader_args(cfg)
+    # 5 warning
+    assert len(record) == 5
+    # assert the warning message
+    assert 'train_dataloader' in record.list[0].message.args[0]
+    assert 'samples_per_gpu' in record.list[0].message.args[0]
+    assert 'persistent_workers' in record.list[1].message.args[0]
+    assert 'train_dataloader' in record.list[1].message.args[0]
+    assert 'workers_per_gpu' in record.list[2].message.args[0]
+    assert 'train_dataloader' in record.list[2].message.args[0]
+    assert cfg.data.train_dataloader.workers_per_gpu == 1
+    assert cfg.data.train_dataloader.samples_per_gpu == 1
+    assert cfg.data.train_dataloader.persistent_workers
+    assert cfg.data.val_dataloader.workers_per_gpu == 1
+    assert cfg.data.val_dataloader.samples_per_gpu == 3
+    assert cfg.data.test_dataloader.workers_per_gpu == 1
+    assert cfg.data.test_dataloader.samples_per_gpu == 2
+
+    # test test is a list
+    cfg = ConfigDict(
+        dict(
+            data=dict(
+                samples_per_gpu=1,
+                persistent_workers=True,
+                workers_per_gpu=1,
+                val=dict(samples_per_gpu=3),
+                test=[dict(samples_per_gpu=2),
+                      dict(samples_per_gpu=3)],
+                train=dict())))
+
+    with pytest.warns(None) as record:
+        cfg = compat_loader_args(cfg)
+    # 6 warning
+    assert len(record) == 6
+    assert cfg.data.test_dataloader.samples_per_gpu == 3
+
+    # assert can not set args at the same time
+    cfg = ConfigDict(
+        dict(
+            data=dict(
+                samples_per_gpu=1,
+                persistent_workers=True,
+                workers_per_gpu=1,
+                val=dict(samples_per_gpu=3),
+                test=dict(samples_per_gpu=2),
+                train=dict(),
+                train_dataloader=dict(samples_per_gpu=2))))
+    with pytest.raises(AssertionError):
+        compat_loader_args(cfg)
+    cfg = ConfigDict(
+        dict(
+            data=dict(
+                samples_per_gpu=1,
+                persistent_workers=True,
+                workers_per_gpu=1,
+                val=dict(samples_per_gpu=3),
+                test=dict(samples_per_gpu=2),
+                train=dict(),
+                val_dataloader=dict(samples_per_gpu=2))))
+    with pytest.raises(AssertionError):
```

Comment:
Add comments to indicate what error is testing, you can use 
`with pytest.raises(AssertionError, match='error message'):`